streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/KTableSuppressProcessor.java

```diff
@@ -77,6 +84,7 @@ public void init(final ProcessorContext context) {
     public void process(final K key, final Change<V> value) {
         buffer(key, value);
         enforceConstraints();
+        suppressionSensor.record();
```

Comment:
This seems to effectively count the number of input records (total and average) -- so it's basically input data rate? Is it intended like this. `intermediate-result-suppression` sounds more like how often something is suppressed, but I don't see how this would be reflected here.

Reply:
I struggled a bit with this question as well. Technically, everything that goes into the buffer is suppressed, and then some stuff is emitted later. We have one metric for each of these things.

I agree it seems more like we should be measuring the number of records that are discarded. I guess we could do this by recording the metric when we buffer a record for which there was a pre-existing record in the buffer. This would be straightforward for the in-memory buffer, but I'm concerned about doing an unnecessary `get` with an on-disk store to maintain the metric.

You should be able to arrive at the same (or similar) number by subtracting the emit metric from the suppress metric. I was thinking this is what people would probably do, but on second though, I could put together a composite metrics that does it internally.

WDYT?

Reply:
Ok, I can get behind this. It should be measuring the same thing as the `process-rate` metric for the suppression ProcessorNode.

I'll remove this metric.